### ISSUE TITLE
Shade blocks according to average light direction in a chunk

### ISSUE THREAD

**Issue by ikabod-kee** [PixelGuys/Cubyz#844]
![image](https://github.com/user-attachments/assets/2f9e25c7-d297-4d7b-b1cf-ce208d30d5ea)

Simulated results:
![image](https://github.com/user-attachments/assets/52bdc030-293e-4c53-b3b3-62b439ae9968)
![image](https://github.com/user-attachments/assets/4f22e5a4-7076-4706-9325-f3ade8b726fa)
![image](https://github.com/user-attachments/assets/cba6cba0-c97b-4ba6-9d8f-7f952b00067d)

**Comment by CtrlAltRat** [comment]
I find that with the current lighting it gets really hard to tell depth. Especially when looking at forest form the top. I have no clue how you would even start to fix this however.

Very helpful issue/comment I know. Was encouraged to put this here.

**Comment by careeoki** [comment]
This solution would create giant shadows under floating islands.
I think #888 is a much better solution for this.

**Comment by IntegratedQuantum** [comment]
Yeah, shadow maps are definitely the way to go when it comes to stuff that's illuminated by the sun.

As for depth perception, I think we are a bit screwed there, since forests, especially the jungle, are fundamentally just random blobs of leaves when looked at from above:

![Image](https://github.com/user-attachments/assets/2b25c0a8-dc83-4aba-8881-e06d0f0bb600)

But to be honest, the situation that you look at a forest from above should be pretty rare in survival, so I don't think we need to design our graphics around that.
